### PR TITLE
Refactor docstrings documentation

### PR DESCRIPTION
**Pull request by elenatorro**
I fixed some little issues related with the Dataset documentation

### CODE REVIEW

File: cartoframes/data/dataset/dataset_info.py [hunk 1]
```diff
@@ -12,8 +12,13 @@
 
 class DatasetInfo(object):
     PRIVATE = 'PRIVATE'
+    """Dataset privacy for datasets that are private"""
+
     PUBLIC = 'PUBLIC'
+    """Dataset privacy for datasets that are public"""
+
     LINK = 'LINK'
+    """Dataset privacy for link that are accessible by link"""
```
Comment (oleurud) [inline cartoframes/data/dataset/dataset_info.py]:
```suggestion
    """Dataset privacy for datasets that are accessible by link"""
```

File: cartoframes/data/dataset/dataset.py [hunk 2]
```diff
@@ -230,7 +218,7 @@ def download(self, limit=None, decode_geom=False, retry_times=DOWNLOAD_RETRY_TIM
 
         return self._strategy.download(limit, decode_geom, retry_times)
 
-    def upload(self, with_lnglat=None, if_exists=FAIL, table_name=None, schema=None, credentials=None):
+    def upload(self, with_lnglat=None, if_exists=BaseDataset.FAIL, table_name=None, schema=None, credentials=None):
```
Comment (oleurud) [inline cartoframes/data/dataset/dataset.py]: I don't see why move constants to `BaseDataset`.  `BaseDataset` is an internal class, a user should never use it. I would prefer to use the constants from `Dataset`.  Maybe I am losing something

File: cartoframes/data/dataset/dataset.py [hunk 3]
```diff
@@ -316,39 +304,44 @@ def delete(self):
...
-        """Checks if the Dataset is local (DataFrameDataset)"""
-        return isinstance(self._strategy, DataFrameDataset)
+        Example:
 
-    def is_remote(self):
-        """Checks if the Dataset is local (TableDataset or QueryDataset)"""
-        return not self.is_local()
+            .. code::
+
+                from cartoframes.data import Dataset, DatasetInfo
```
[the remark this answers is not in the document]
Comment (elenatorro) [reply cartoframes/data/dataset/dataset.py]: I added it cause it was also referenced in the documentation, should we remove then the `DatasetInfo` class from the init?

File: cartoframes/data/dataset/dataset.py (hunk 2 above)
Comment (elenatorro) [reply cartoframes/data/dataset/dataset.py]: 👍 In this case, let's move it back to Dataset but then they should be documented

**Pull request merged**